daemon/protocol/python/scoot/scoot.py

```diff
@@ -0,0 +1,199 @@
+  
+  
+def run(cmd):
+  #verfiy args
+  timeout = cmd['--timeout']
+  if not timeout:
+    timeout_ns = int(1e9)
+  else:
+    try:
+      timeout_ns = int(int(timeout) * 1e9)
+    except Exception as e:
+      raise docopt.DocoptExit("invalid value for timeout, must be an integer or decimal number.")
+  snapshotId = cmd['--snapshotId']
+  args = cmd['<command>']
+
+  #run the command
+  try:
+    runId = proto.run(snapshot_id=snapshotId, argv=args, timeout_ns=timeout_ns)
+    #print the output
+    print("run id = {0}".format(runId))
+  #handle errors running the command
+  except proto.ScootException as e:
+    if re.search("No resources available", e) is None:
+      print(e)
+    elif re.search("Not started", e) > 0 or re.search("UNAVAILABLE", e) > 0:
+      raise docopt.DocoptExit("Scoot Daemon is not running!\n")
+    else:
+      raise docopt.DocoptExit("run request error: '{0}'".format(str(e)))
+  except Exception as e1:
+    raise docopt.DocoptExit("run request error:'{0}'".format(str(e1)))
+
+
+def poll(cmd):
+  #verify args
+  wait = cmd["--wait"]
+  if not wait:
+    wait = 0
+  try:
+    wait = int(wait)
+  except Exception as e:
+    raise docopt.DocoptExit("Wait must be an integer.")
+  runIds = cmd["<runId>"]
+  if not cmd["<runId>"]:
+    raise docopt.DocoptExit("RunId list is empty!")
+  
+  #run the command
+  try:
+    statuses = proto.poll(run_ids=cmd["<runId>"], timeout_ns=wait, return_all=cmd['--all'])
+    # print the output
+    display_statuses(statuses)
+  #handle errors running the command
+  except proto.ScootException as e:
+    if re.search("Not started", e) > 0 or re.search("UNAVAILABLE", e) > 0:
+      raise docopt.DocoptExit("Scoot Daemon is not running!\n")
+    raise docopt.DocoptExit("poll request error:'{0}'.".format(str(e)))
+
+
+def echo(cmd):
+  #verify args
+  if not cmd['<ping>']:
+    raise docopt.DocoptExit("Must supply <ping> value to echo.")
+  
+  #run the command
+  try:
+    echo = proto.echo(ping=cmd['<ping>'])
+    #print the output
+    print("{0}".format(echo)) 
+  #handle errors running the command
+  except proto.ScootException as e:
+    if re.search("Not started", str(e)) > 0 or re.search("UNAVAILABLE", str(e)) > 0:
+      raise docopt.DocoptExit("Scoot Daemon is not running!\n")
+    raise docopt.DocoptExit("echo request error:'{0}'".format(str(e)))
+
+
+if __name__ == '__main__':
+  # parse the command line
+  try:
+    cmd = docopt.docopt(__doc__)
+  except Exception as e:
+    raise docopt.DocoptExit("error parsing command: '{0}'.".format(str(e)))
+  
+  # make the client connection
+  try:
+    proto.start()
+  #handle errors making the client connection
+  except proto.ScootException as e:
+    if re.search("UNAVAILABLE", e) > 0:
+      raise docopt.DocoptExit("Scoot Daemon is not running!\n")
+    raise docopt.DocoptExit("connecting to daemon error: '{0}' (make sure you have started the daemon).".format(str(e)))       
```

Comment:
Again not sure why we would re-raise as a docopt error. I would think DocoptExit would be for command parsing errors, this is a scoot error